src/camera.rs

```diff
@@ -18,31 +18,34 @@ impl Camera {
         }
         .rotate(self.transform.rotation);
 
-        let tris = world
-            .tris
+        world
+            .objects
             .iter()
-            .filter_map(|p| self.tri_raycast(ray, world.light, *p));
-        let spheres = world
-            .spheres
-            .iter()
-            .filter_map(|p| self.sphere_raycast(ray, world.light, *p));
-        tris.chain(spheres)
+            .filter_map(|p| self.raycast(ray, world.light, p))
             .min_by(|(_, a), (_, b)| a.total_cmp(b))
             .map(|(color, _)| color)
             .unwrap_or(Color([0; 3]))
     }
 
+    fn raycast(&self, ray: Vec3, light: Vec3, p: &Object) -> Option<(Color, f32)> {
+        match *p {
```

Comment:
`p` wasa originally named as such for "parameters" but I think now we can change it to `obj` or `o` for "object."
```suggestion
            .filter_map(|obj| self.raycast(ray, world.light, obj))
            .min_by(|(_, a), (_, b)| a.total_cmp(b))
            .map(|(color, _)| color)
            .unwrap_or(Color([0; 3]))
    }

    fn raycast(&self, ray: Vec3, light: Vec3, obj: &Object) -> Option<(Color, f32)> {
        match *obj {
```